src/term_ctl.c

```diff
@@ -0,0 +1,80 @@
+/**
+ * Terminal control utility functions.
+ *
+ * Copyright (C) 2018 Christian Zuckschwerdt
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ */
+
+#include <stdio.h>
+#include <unistd.h>
+#ifndef _WIN32
+#include <sys/ioctl.h>
+#endif
+
+#include "term_ctl.h"
+
+int term_get_columns(int fd)
+{
```

Comment:
This is called once per data output to determine the available width for columns. If this is much too expensive then we can put that in init, but it wouldn't adapt to terminal resize anymore.
I don't think data output should come at more than a few per second, say 3.

Reply:
FWIW, on POSIX you can listen for the signal SIGWINCH to avoid polling for size changes; [apparently Windows also has a solution](https://stackoverflow.com/questions/10856926/sigwinch-equivalent-on-windows). I don't know whether it's worth doing here, though; the usual use case would be triggering redraw in a full-screen application like an editor.

Reply:
We just need the current value when we output data. I measured the execution time at 1 µs, which seems no slower than a few printfs. Neglible to do a few times per second at most. Can't say about Windows though.

Reply:
Okay, I suspect it's equally negligible times for a [GetConsoleScreenBufferInfo()](https://docs.microsoft.com/en-us/windows/console/getconsolescreenbufferinfo).